python/spear/env.py

```diff
@@ -86,10 +86,10 @@ def close(self):
         self._observation_space_info.terminate()
         self._task_step_info_space_info.terminate()
         self._agent_step_info_space_info.terminate()
-
-        self._close()
-        self._wait_until_unreal_instance_is_closed()
-        self._close_client()
+        if self._config.SPEAR.LAUNCH_MODE != "running_instance":
```

Comment:
Refactor as follows so that the there are symmetric `request_launch_unreal_instance` and `request_close_unreal_instance` functions with matching if statements in each. In turn, this will make the constructor more symmetrical with the `close()` function.

```
def request_close_unreal_instance():
    if self._config.SPEAR.LAUNCH_MODE != "running_instance":
        self._close()
        self._wait_until_unreal_instance_is_closed()
```

Now the `close()` function can be implemented as follows, which matches the pattern in the constructor:

```
    def close(self):
        # ...

        self._action_space_info.terminate()
        self._observation_space_info.terminate()
        self._task_step_info_space_info.terminate()
        self._agent_step_info_space_info.terminate()

        # Note that in the constructor, we launch the Unreal instance first and then initialize the client. Normally, we
        # would do things in the reverse order here. But if we close the client first, then we can't send a command to
        # the Unreal instance to close it. So we close the Unreal instance first and then close the client.
        self.request_close_unreal_instance()
        self._close_client()
```

Reply:
fixed